TrackletAlgorithm/VMRouterCMTop.h

```diff
@@ -0,0 +1,83 @@
+#ifndef TrackletAlgorithm_VMRouterCMTop_h
+#define TrackletAlgorithm_VMRouterCMTop_h
+
+#include "VMRouterCM.h"
+
+// VMRouter Top Function for layer 2, AllStub region A
+// Sort stubs into smaller regions in phi, i.e. Virtual Modules (VMs).
+
+// NOTE: to run a different phi region, change the following
+//          - constants specified in VMRouterCMTop.h
+//          - add/remove pragmas depending on inputStubs in VMRouterCMTop.cc
+//          - maskASI in VMRouterCMTop.cc
+//          - add the phi region in emData/download.sh, make sure to also run clean
+
+
+////////////////////////////////////////////
+// Variables for that are specified with regards to the VMR region
+// Changed manually
+
+#define kLAYER 2 // Which barrel layer number the data is coming from
+#define kDISK 0 // Which disk number the data is coming from, 0 if not disk
+
+constexpr char phiRegion = 'A'; // Which AllStub/PhiRegion
+constexpr int sector = 4; //  Specifies the sector
+
+// Maximum number of memory "copies" for this Phi region
+constexpr int numASCopies(4); // Allstub memory
+constexpr int numASInnerCopies(4); // Allstub memory Note: can't use 0 if we don't have any memories of a certain type. Use 1.
+constexpr int numTEOCopies(3); // TE Outer memories, can be 0 when no TEOuter memories
+
+// Number of inputs
+constexpr int numInputs(2); // Number of input memories, EXCLUDING DISK2S
+constexpr int numInputsDisk2S(0); // Number of DISK2S input memories
```

Comment:
I guess these are not actually free parameters, but instead are functions of the layer/disk and phi region. So would it make sense to define them as `constexpr` functions that take in the layer/disk and phi region as arguments or template parameters? You could define the functions for just VMRCM_L2PHIA for now, and then expand it to other instances of the combined VMR in the future, but this way, only a couple parameters would need to change in order to switch between VMR instances.

Reply:
Not quite sure I understand. Do you want a function that computes the number of copies somehow, or more like a LUT? Because I have no idea how one determines how many memories one needs, I just look at the wiring (and/or what was being downloaded)...

Reply:
I mean something like what's done for the TP in TrackletProcessor_parameters.h or for the TC in TrackletCalculator_parameters.h. These functions aren't doing any real computation, they just return values that are hard-coded based on the wiring. But the point is that once these functions are eventually set up for all instances, only the layer/disk and phi region would have to be changed, and not all these additional constants.